client/src/main/java/org/eclipse/hono/client/impl/AmqpAdapterClientTelemetrySenderImpl.java

```diff
@@ -0,0 +1,123 @@
+     */
+    public static Future<TelemetrySender> createWithAnonymousLinkAddress(
+            final HonoConnection con,
+            final String tenantId,
+            final Handler<String> remoteCloseHook) {
+
+        Objects.requireNonNull(con);
+        Objects.requireNonNull(tenantId);
+
+        final String targetAddress = null; // anonymous relay
+        return con.createSender(targetAddress, ProtonQoS.AT_LEAST_ONCE, remoteCloseHook)
+                .map(sender -> new AmqpAdapterClientTelemetrySenderImpl(con, sender, tenantId, targetAddress));
+    }
+
+    @Override
+    public Future<ProtonDelivery> send(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties) {
+        return send(deviceId, payload, contentType, properties, null);
+    }
+
+    @Override
+    public Future<ProtonDelivery> send(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties, final SpanContext context) {
+        final Message msg = createMessage(deviceId, payload, contentType, properties);
+        return send(msg, context);
+    }
+
+    @Override
+    public Future<ProtonDelivery> sendAndWaitForOutcome(final String deviceId, final byte[] payload,
+            final String contentType, final Map<String, ?> properties) {
+        return sendAndWaitForOutcome(deviceId, payload, contentType, properties, null);
+    }
+
+    @Override
+    public Future<ProtonDelivery> sendAndWaitForOutcome(final String deviceId, final byte[] payload,
+            final String contentType, final Map<String, ?> properties, final SpanContext context) {
+        final Message msg = createMessage(deviceId, payload, contentType, properties);
+        return sendAndWaitForOutcome(msg, context);
+    }
+
+    private Message createMessage(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties) {
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(payload);
+        Objects.requireNonNull(contentType);
+
+        final Message msg = ProtonHelper.message();
+        msg.setAddress(getTo(deviceId));
+        MessageHelper.setPayload(msg, contentType, payload);
+        setApplicationProperties(msg, properties);
+        MessageHelper.addDeviceId(msg, deviceId);
+        return msg;
+    }
+
+    /**
+     * Shadowing the static factory method that must not be used outside a protocol adapter.
+     *
+     * @param con not used.
+     * @param tenantId not used.
+     * @param remoteCloseHook not used.
+     * @return nothing.
+     * @throws UnsupportedOperationException always.
+     */
+    public static Future<DownstreamSender> create(
+            final HonoConnection con,
+            final String tenantId,
+            final Handler<String> remoteCloseHook) {
+        throw new UnsupportedOperationException("This method must not be used outside a protocol adapter");
```

Comment:
IMHO, shadowing the (unsupported) parent static factory method with what is now in `createWithAnonymousLinkAddress` would be a better idea (i.e. renaming `createWithAnonymousLinkAddress` to just `create`). WDYT?